### PR TITLE
[ENH] update default colors

### LINKED ISSUE #3979: plots in plot_second_level_association_test.py don't show any activation
> * Last 2 plots in https://nilearn.github.io/dev/auto_examples/05_glm_second_level/plot_second_level_association_test.html don't show any contrasts. Is that as expected?

checked back all the way to version 0.8.0 this problem is still there

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/issues/3728#issuecomment-1720056866_

For comparison here is the original nistats example: https://nistats.github.io/auto_examples/03_second_level_models/plot_second_level_association_test.html#sphx-glr-auto-examples-03-second-level-models-plot-second-level-association-test-py

Showing no activation feels a bit underwhelming, no? Should we then just amend this example?

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes to https://github.com/nilearn/nilearn/issues/4848
- closes #3979 

Changes proposed in this pull request:

- udpate defaults colors
  - makes RdBu_r the default in most cases
  - use pure gray scale for some (plot_anat, plot_epi...)
- fix most examples to use the proper color scale when using non divergent data

### CODE REVIEW

File: examples/01_plotting/plot_dim_plotting.py [hunk 1]
```diff
@@ -29,12 +29,14 @@
 # Plotting with enhancement of background image with dim=-.5
 # ----------------------------------------------------------
 
-from nilearn import plotting
+from nilearn.plotting import plot_stat_map, show
 
-plotting.plot_stat_map(
+cut_coords = (36, -27, 66)
```
Comment (Remi-Gau) [inline examples/01_plotting/plot_dim_plotting.py:34]: - changing of coordinates closes https://github.com/nilearn/nilearn/issues/3979

File: examples/02_decoding/plot_haxby_glm_decoding.py [hunk 2]
```diff
@@ -149,21 +149,21 @@
...
-#       problem keeping only 5% of voxels which are most informative.
+# * we use univariate feature selection to reduce the dimension of the
+#   problem keeping only 5% of voxels which are most informative.
 #
-#     * a cross-validation scheme, here we use LeaveOneGroupOut
-#       cross-validation on the runs which corresponds
-#       to a leave-one-run-out
+# * a cross-validation scheme, here we use LeaveOneGroupOut
+#   cross-validation on the runs which corresponds
+#   to a leave-one-run-out
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_glm_decoding.py:166]: just to improve rendering of the list

File: examples/05_glm_second_level/plot_second_level_association_test.py [hunk 3]
```diff
@@ -86,12 +86,16 @@
 # Let us plot the second level :term:`contrast` at the computed thresholds.
 from nilearn.plotting import plot_stat_map, show
 
+cut_coords = [10, -5, 10]
```
Comment (Remi-Gau) [inline examples/05_glm_second_level/plot_second_level_association_test.py:89]: - changing of coordinates closes https://github.com/nilearn/nilearn/issues/3979

File: examples/07_advanced/plot_bids_analysis.py [hunk 4]
```diff
@@ -131,13 +132,14 @@
...
-        colorbar=False,
+        colorbar=True,
         threshold=p001_unc,
         title=f"sub-{model.subject_label}",
         axes=axes[int(midx / ncols), int(midx % ncols)],
         plot_abs=False,
         display_mode="x",
-        cmap="bwr",
+        vmin=-12,
+        vmax=12,
```
Comment (Remi-Gau) [inline examples/07_advanced/plot_bids_analysis.py:142]: to make it easier to compare plotted subjects

**Comment by Remi-Gau** [comment]
this PR adds a lot of small  edits to several examples: very often just adding a `show()` to make it easier to view figures when running locally

**Review by bthirion — COMMENTED** [review]
Happy with the change, but it deserves a watsnew entry, as it changes defaults in image plotting.

**Comment by Remi-Gau** [comment]
Yup definitely.
I tried to be thorough bu I may have missed a few things so this PR may have children in the future.

**Review by bthirion — COMMENTED** [review]
LGTM still.

File: examples/02_decoding/plot_oasis_vbm.py [hunk 5]
```diff
@@ -60,7 +60,7 @@
 from nilearn.image import get_data
 from nilearn.maskers import NiftiMasker
 
-n_subjects = 100  # more subjects requires more memory
+n_subjects = 50  # more subjects requires more memory
```
Comment (bthirion) [inline examples/02_decoding/plot_oasis_vbm.py:63]: oops. I understand the concern regarding light models, but we're giving a poor example, doing ML with low n.

File: examples/02_decoding/plot_oasis_vbm_space_net.py [hunk 6]
```diff
@@ -17,7 +19,9 @@
 
 from nilearn import datasets
 
-n_subjects = 200  # increase this number if you have more RAM on your box
+# Number of subjects is kept low for the sake of speed.
+# Increase this number if you have more RAM on your computer.
+n_subjects = 50
```
Comment (bthirion) [inline examples/02_decoding/plot_oasis_vbm_space_net.py:24]: Same thing here. I'm afraid of making the results dicouragingly poor.

**Review by bthirion — COMMENTED** [review]
Looks good, but we need to draw a compromise btw resource consumption and pedagogy.

File: examples/02_decoding/plot_oasis_vbm.py (hunk 5 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm.py:63]: Good with me: not actually 100% it speeds things up and in any case this is beyond the scope of this PR.

There are a few other places (in previous PRs) where I reduced the number of subjects included in the example: should I plan to revert those?

Comment (bthirion) [reply examples/02_decoding/plot_oasis_vbm.py]: If it's not justified by significant computation time savings, I'd revert if you don't mind.

Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm.py]: will fix in a separate PR

File: examples/00_tutorials/plot_decoding_tutorial.py [hunk 7]
```diff
@@ -273,7 +283,7 @@
 )
 decoder.fit(fmri_niimgs, conditions, groups=run_label)
 
-print(decoder.cv_scores_)
+print(f"{decoder.cv_scores_=}")
```
Comment (man-shu) [inline examples/00_tutorials/plot_decoding_tutorial.py:286]: oh nice didn't know about this one

File: examples/00_tutorials/plot_single_subject_single_run.py [hunk 8]
```diff
@@ -248,7 +248,7 @@
     two_sided=False,  # using a one-sided test
 )
 # Let's use a sequential colormap as we will only display positive values.
-plotting_config["cmap"] = "black_red"
+plotting_config["cmap"] = "inferno"
```
Comment (man-shu) [inline examples/00_tutorials/plot_single_subject_single_run.py:251]: Reading the changelog I thought we were updating to RdBu_r

File: examples/01_plotting/plot_demo_glass_brain_extensive.py [hunk 9]
```diff
@@ -101,7 +102,7 @@
     vmin=0,
     threshold=2,
     symmetric_cbar=False,
-    cmap="viridis",
+    cmap="inferno",
```
Comment (man-shu) [inline examples/01_plotting/plot_demo_glass_brain_extensive.py:105]: Same in this file

File: doc/changes/latest.rst [hunk 10]
```diff
@@ -16,3 +16,5 @@ Enhancements
 
 Changes
 -------
+
+- :bdg-info:`Plotting` Change the default map to be ``"RdBu_r"`` or ``"gray"`` for most plotting functions (:gh:`4807` by `Rémi Gau`_).
```
Comment (man-shu) [inline doc/changes/latest.rst:20]: Looks like we have used inferno in more places than RdBu_r. I guess this that's because they look similar?

**Review by man-shu — APPROVED** [review]
LGTM

File: doc/changes/latest.rst (hunk 10 above)
Comment (Remi-Gau) [reply doc/changes/latest.rst:20]: I think this impression comes from the fact that I have 'fixed' / 'updated' the examples to use a proper sequential color map.

There a were a lot of examples that used the default diverging colormap when a sequential colormap was actually better suited.

Will update the changelog.

Comment (Remi-Gau) [inline doc/changes/latest.rst:20]:
```suggestion
- :bdg-info:`Plotting` Change the default map to be ``"RdBu_r"`` or ``"gray"`` for most plotting functions. In several examples, use the "inferno" colormap when a sequential colormap is preferable (:gh:`4807` by `Rémi Gau`_).
```

**Pull request merged**